miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/PoolRestController.java

```diff
@@ -279,6 +283,46 @@ public PoolPickerResponse getPickersBySearch(@RequestParam("platform") String pl
         PaginationFilter.platformType(PlatformType.valueOf(platform)));
   }
 
+  @RequestMapping(value = "avginsertsizes", method = RequestMethod.POST, produces = "application/json")
+  @ResponseBody
+  public JSONObject calculateAverageInsertSizes(@RequestBody JSONObject json) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)